src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
@HeartSaVioR in this case it's ok if the connection is not marked broken because in the original issue #1168 @serge-groshev was getting a JedisDataException because Redis was still loading the dataset into memory, so it's OK to re-use the same connection afterwards when retrieving another connection. 

Regarding the IllegalArgumentException as far as I can remember it can only happen when you try to return an resource which doesn't belong to the pool which can't be this case.